File: backend/src/main/java/com/NBE_4_5_2/Team5/domain/user/admin/dto/NoticeResBody.java
package com.NBE_4_5_2.Team5.domain.user.admin.dto;

import java.time.LocalDateTime;

import com.NBE_4_5_2.Team5.domain.user.admin.entity.NoticePost;

public record NoticeResBody(String id, String title, String content, AdminResBody admin, LocalDateTime createdAt) {
	public static NoticeResBody of(NoticePost saved) {
		return new NoticeResBody(saved.getId(), saved.getTitle(), saved.getContent(),
			new AdminResBody(saved.getAdmin()), saved.getCreatedAt());
	}
}

File: backend/src/main/java/com/NBE_4_5_2/Team5/domain/user/admin/dto/NoticeResBody.kt
package com.NBE_4_5_2.Team5.domain.user.admin.dto

import com.NBE_4_5_2.Team5.domain.user.admin.entity.NoticePost
import java.time.LocalDateTime

data class NoticeResBody(
	val id: String,
	val title: String,
	val content: String,
	val admin: AdminResBody,
	val createdAt: LocalDateTime
) {
    companion object {
        @JvmStatic
		fun of(saved: NoticePost): NoticeResBody {
            return NoticeResBody(
                saved.id, saved.title, saved.content,
                AdminResBody(saved.admin), saved.createdAt
            )
        }

    }
}
